src/Main/Main.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+/**
+ * The file that defines the main start class.
+ *
+ * A class definition that includes attributes and functions used across both the
+ * theme-facing side of the site and the admin area.
+ *
+ * @package EightshiftForms\Main
+ */
+
+declare(strict_types=1);
+
+namespace EightshiftForms\Main;
+
+use EightshiftLibs\Main\AbstractMain;
+use EightshiftForms\Cache;
+use EightshiftForms\Captcha;
+use EightshiftForms\Rest;
+use EightshiftForms\Integrations;
+use EightshiftFormsTests\Mocks;
+
+/**
+ * The main start class.
+ *
+ * This is used to define admin-specific hooks, and
+ * theme-facing site hooks.
+ *
+ * Also maintains the unique identifier of this theme as well as the current
+ * version of the theme.
+ */
+class Main extends AbstractMain
+{
+
+	/**
+	 * Set this to true if you need dependency injection in tests.
+	 *
+	 * @var boolean
+	 */
+	private $isTest = false;
+
+	/**
+	 * Register the project with the WordPress system.
+	 *
+	 * The registerService method will call the register() method in every service class,
+	 * which holds the actions and filters - effectively replacing the need to manually add
+	 * them in one place.
+	 *
+	 * @return void
+	 */
+	public function register(): void
+	{
+		\add_action('after_setup_theme', [$this, 'registerServices']);
+	}
+
+	/**
+	 * Provides additional / different services depending on if we're in test or not.
+	 *
+	 * @param   boolean $isTest Set to true if running tests.
+	 * @return void
+	 */
+	public function setTest(bool $isTest): void
+	{
+		$this->isTest = $isTest;
+	}
+
+	/**
+	 * Get the list of services to register.
+	 *
+	 * A list of classes which contain hooks.
+	 *
+	 * @return array<int|string, array<int, class-string>|class-string> Array of fully qualified service class names.
+	 */
+	protected function getServiceClasses(): array
+	{
+		return $this->isTest ? $this->getTestServiceClasses() : $this->getProdServiceClasses();
+	}
+
+
+	/**
+	 * Get the list of services to register (for production).
+	 *
+	 * A list of classes which contain hooks.
+	 *
+	 * @return array<int|string, array<int, class-string>|class-string> Array of fully qualified service class names.
+	 */
+	protected function getProdServiceClasses(): array
+	{
+		return [];
+	}
+
+	/**
+	 * Get the list of services to register (for testing).
+	 *
+	 * A list of classes which contain hooks.
+	 *
+	 * @return array<int|string, array<int, class-string>|class-string> Array of fully qualified service class names.
+	 */
+	protected function getTestServiceClasses(): array
+	{
+		/**
+		 * Ignoring this rule because this is how PHP-DI expects the dependency tree to be defined.
+		 */
+		/* phpcs:disable Universal.Arrays.MixedKeyedUnkeyedArray.Found */
```

Comment:
Why is this silenced? What error are you getting here?

Reply:
`Universal.Arrays.MixedKeyedUnkeyedArray.Found`

It complains because of the different way classes with/without dependencies are defined.

```
  SomeClassNoDeps::class,
  SomeClassWithDeps::class => [
    DepClass: class
  ]
```